--- /tmp/va-orig/yaAGC/Makefile	2021-05-21 07:45:24.000000000 +0000
+++ /tmp/virtualagc-ddc65e7bed41f1301921b934fcbaaee93db99dda/yaAGC/Makefile	2026-07-29 15:37:43.872874181 +0000
@@ -145,7 +145,7 @@
 LDFLAGS2 = \
 	--no-entry \
 	--export-dynamic \
-	--unresolved-symbols=import-functions \
+	--allow-undefined \
 	-L ${WASI_SDK_PATH}/share/wasi-sysroot/lib/wasm32-wasi \
 	-lc \
 	--export=malloc \
@@ -217,6 +217,23 @@
 	wasm-opt $@ -Oz -o $@
 	wasm-strip $@
 
+
+yaAGC-ext.wasm: wasm.o hwio.o agc_engine.o agc_engine_init.o agc_utilities.o ringbuffer_api.o ringbuffer.o version.o
+	${WASI_SDK_PATH}/bin/wasm-ld ${LDFLAGS2} -o $@ $^ \
+	  --export=agc_counter_increment \
+	  --export=agc_hw_input_apply \
+	  --export=agc_hw_input_last_error_index \
+	  --export=agc_out_trace_entry_size \
+	  --export=agc_out_trace_drain \
+	  --export=agc_out_trace_dropped \
+	  --export=agc_out_trace_reset \
+	  --export=agc_out_trace_set_enabled \
+	  --export=agc_out_trace_enabled \
+	  --export=agc_ext_version \
+	  --export=agc_hwio_version
+	wasm-opt $@ -Oz -o $@
+	wasm-strip $@
+
 yaAGC.wast: yaAGC.wasm
 	wasm2wat $< > $@
 
